rust/private/rustc.bzl

```diff
@@ -320,17 +320,34 @@ def rustc_compile_action(
     # and use `$(pwd)` which resolves the `exec_root` at action execution time.
     package_dir = ctx.build_file_path[:ctx.build_file_path.rfind("/")]
     manifest_dir_env = "CARGO_MANIFEST_DIR=$(pwd)/{} ".format(package_dir)
-    command = '{}{}{} "$@" --remap-path-prefix="$(pwd)"=__bazel_redacted_pwd'.format(
-        manifest_dir_env,
-        out_dir_env,
-        toolchain.rustc.path,
+    maybe_rename = ""
+    if crate_info.type == "bin":
+        generated_file = crate_info.name
+        if toolchain.target_arch == "wasm32":
+            generated_file = generated_file + ".wasm"
+        src = "/".join([output_dir, generated_file])
+        dst = crate_info.output.path
+        if src != dst:
+            maybe_rename = " && /bin/mv {src} {dst}".format(src=src, dst=dst)
```

Comment:
Why do you need that maybe_rename? I cannot make sense of the logic.

Reply:
If a target name is `foo-bar`, both Cargo and rules_rust will translate this into a crate name of `foo_bar`, and so `rustc` will create an output file (in an intermediate directory) named `foo_bar`.

However, Cargo, when putting the output in your `target/${config}` directory will use the original target name of `foo-bar`.

The env var Cargo sets uses uses the original target name, with `-` not `_`.

If we'd rather not do this, we could instead just set the env var to use `foo-bar` and point at a file named `foo_bar`, but this seemed more compatible with what Cargo does.

If we want to keep doing this, I can gladly add more comments to explain :)

Reply:
I am fine with this behavior but I'd gladly take some comment here :)

Reply:
Added comment